clang/test/C/C2y/n3369.c

```diff
@@ -0,0 +1,68 @@
+// RUN: %clang_cc1 -fsyntax-only -std=c2y -pedantic -Wall -Wno-comment -verify %s
+
+/* WG14 N3369: Clang 21
+ * _Lengthof operator
+ *
+ * Adds an operator to get the length of an array. Note that WG14 N3469 renamed
+ * this operator to _Countof.
+ */
```

Comment:
More tests for you to consider:

 - unspecified arrays length
 - arrays of rank 3
 - `_Countof array`
 - typedefs
 - `static_assert(__is_same_as(typeof(_Countof(foo)), size_t))`
 - Constant folding tests

Reply:
Good call, I'll add these tests. FWIW, I already show constant folding:
```
  // Constant folding works the same way as sizeof, too.
  const int m = 12;
  int other_array[m];
  static_assert(sizeof(other_array));   // expected-error {{static assertion expression is not an integral constant expression}}
  static_assert(_Countof(other_array)); // expected-error {{static assertion expression is not an integral constant expression}}
  static_assert(sizeof(int[m]));        // expected-error {{static assertion expression is not an integral constant expression}}
  static_assert(_Countof(int[m]));      // expected-error {{static assertion expression is not an integral constant expression}}
```
or did you have something else in mind?

Reply:
I'm blind!

Reply:
Can you make `m` non-const for added clarity?

Reply:
Then it wouldn't be testing constant folding, so I feel like I must be misunderstanding something.